src/haddock/setup.py

```diff
@@ -0,0 +1,71 @@
+import toml
+import shutil
+import copy
+from pathlib import Path
+from haddock.error import SetupError
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Setup:
```

Comment:
It might be okay to have a file named `setup.py`, though I dislike it a bit. But the `class Setup` is too general and does not specify what it does. But I understand your intentions. I would suggest a more pipeline based approach. Let me work on that.